File: src/types/PromptModelVersion.ts
// Interfaces for PromptModelVersion update and retrieval processes

/**
 * Interface for the request to read a PromptModel's PromptModelVersions.
 */
export interface ReadPromptModelVersionsRequest {
  prompt_model_uuid: string;
}

/**
 * Interface for the request to read a PromptModelVersion's information.
 */
export interface ReadPromptModelVersionRequest {
  uuid: string;
}

/**
 * Interface for the request to update the published PromptModelVersion.
 */
export interface UpdatePublishedPromptModelVersionRequest {
  uuid: string; // PromptModelVersion UUID
  project_uuid: string;
  project_version: number;
  previous_published_version_uuid: string;
}

/**
 * Interface for the request to update a PromptModelVersion's tags.
 */
export interface UpdatePromptModelVersionTagsRequest {
  uuid: string;
  tags: Array<string>;
}

/**
 * Interface for the request to update a PromptModelVersion's memo.
 */
export interface UpdatePromptModelVersionMemoRequest {
  uuid: string;
  memo: string;
}

/**
 * General interface for representing a PromptModelVersion in the system.
 */
export interface PromptModelVersion {
  id: number;
  created_at: string;
  uuid: string;
  version: number;
  model: string;
  is_published: true;
  from_version: number;
  parsing_type: string;
  output_keys: Array<string>;
  functions: Array<string>;
  is_ab_test: true;
  ratio: number;
  tags: Array<string>;
  memo: string;
  prompt_model_uuid: string;
}

File: api/web/models/prompt_model_version.py
"""APIs for PromptModelVersion"""
from typing import List, Optional

from modules.types import PMObject


class PromptModelVersionInstance(PMObject):
    id: int
    created_at: str
    uuid: str

    version: int
    model: str
    is_published: bool

    from_version: Optional[int] = None
    parsing_type: Optional[str] = None
    output_keys: Optional[List[str]] = None

    functions: Optional[List[str]] = None
    is_ab_test: Optional[bool] = None
    ratio: Optional[float] = None
    tags: Optional[List[str]] = None
    memo: Optional[str] = None

    prompt_model_uuid: str


class UpdatePublishedPromptModelVersionBody(PMObject):
    project_uuid: str
    project_version: int
    previous_published_version_uuid: Optional[str] = None


class UpdatePromptModelVersionTagsBody(PMObject):
    tags: Optional[List[str]] = None
